tests/test_io.py

```diff
@@ -1695,6 +1714,80 @@ def test_csv_options_cs2(self):
 
 class TestResultDymolaBinary:
 
+    def test_next_start_index(self):
+        """
+            Test that calculation of the next start index works as expected.
+
+            This test sets up a dummy FMU and dummy trajectories since we need
+            trajectories of uneven lengths.
+
+        """
+        # Begin by setting up minimal required environment in order to perform the test
+        fmu = Dummy_FMUModelME2([], os.path.join(file_path, "files", "FMUs", "XML", "ME2.0", "CoupledClutches.fmu"),
+            _connect_dll=False)
+
+        result_handler = ResultHandlerBinaryFile(fmu)
+
+        opts = fmu.simulate_options()
+        opts["result_handling"] = "binary"
+        opts["result_handler"] = result_handler
+
+        fmu.setup_experiment()
+        fmu.initialize()
+        opts["initialize"] = False
+
+        result_handler.set_options(opts) # required in order to call simulation_start()
+        result_handler.initialize_complete()
+        result_handler.simulation_start()
+
+        fmu.set('J4.phi', 1) # arbitrary
+        result_handler.integration_point()
```

Comment:
I think that we should test getting the trajectories (using get_variables_data) before we have a call to the integration_point method.

Reply:
`ResultDymolaBinary` constructor fails if you invoke it without calling `integration_point()` before. Do you have any ideas of how to get around this? We currently get an uncaught exception pointing to issues when setting up `self._data_2`.

Reply:
Ok, I think we should fix that. But we can leave that for a follow-up ticket.